### PR TITLE
Moved/Split platform-specific files into platform-specific directories

### PR DESCRIPTION
**Pull request by drichmond**
In preparation for Verilator/BP I've started to move platform-specific files into platform-specific directories. 

The tl;dr of this is that each platform will have its own `hardware.mk` file that defines any platform-specific HDL and `link.mk` that defines how linking is done on the host. 

Passes regression on timing_v0_8_4 and 4x4_fast_n_fake.

* libbsg_manycore_runtime.so is now THE dependency for all linking
* Moved aws.vcs.f (AWS VCS filelist) to platforms/aws-vcs
* Moved gen_simlibs.tcl to platforms/aws-vcs
* Moved contents of testbenches/link.mk to platforms/aws-vcs/link.mk
* Moved msg_config.mk to platforms/aws-vcs
* Modified dramsim3.mk, libdmamem.mk to add dependencies and append to LDFLAGS to libbsg_manycore_runtime instead of LDFLAGS (for application linking) and SIMLIBS
* Included dramsim3.mk, libdmamem.mk, and libinfmem.mk in $(BSG_PLATFORM_PATH)/library.mk
* Moved contents of simlibs.mk to platforms/aws-vcs/hardware.mk
* hardware.mk now includes $(BSG_PLATFORM_PATH)/hardware.mk

### CODE REVIEW

File: hardware/hardware.mk [hunk 1]
```diff
@@ -55,12 +55,6 @@ ifndef BASEJUMP_STL_DIR
 $(error $(shell echo -e "$(RED)BSG MAKE ERROR: BASEJUMP_STL_DIR is not defined$(NC)"))
 endif
 
-# XILINX_VIVADO is set by Vivado's configuration script. We use this
-# as a quick check instead of running Vivado.
-ifndef XILINX_VIVADO
-$(error $(shell echo -e "$(RED)BSG MAKE ERROR: XILINX_VIVADO environment variable undefined. Are you sure Vivado is installed?$(NC)"))
-endif
-
```
Comment (drichmond) [inline hardware/hardware.mk:63]: This check is now platform-specific

**Comment by drichmond** [comment]
Semi-related, my plan is to move `libinfmem.mk`, `libdma.mk` and `libdramsim3.mk` into feature-specific folders like in #584, but this PR seemed easier to do first.

**Comment by drichmond** [comment]
on cosimulation. I'm bugswatting in the HDL by compiling the FPGA image

You'd be amazed how much VCS lest past it.

**Comment by dpetrisko** [comment]
Yep, my experience is VCS > Verilator > Vivado in terms of permissiveness

**Comment by drichmond** [comment]
Can VCS infer parameters? Cosimulation is passing, but a parameter wasn't defined - worse, the wrong name was used to pass it in, and it worked?

**Comment by drichmond** [comment]
This PR compiled the FPGA bitstream, and passed regression on F1 (with the new libraries)

All tests except test_rom passed.

File: libraries/platforms/aws-vcs/hardware.mk [hunk 2]
```diff
@@ -65,78 +55,108 @@ endif
...
 
-VSOURCES += $(BASEJUMP_STL_DIR)/bsg_misc/bsg_cycle_counter.v
-
-# Test DRAM
+# DMA Interface
 VSOURCES += $(BASEJUMP_STL_DIR)/bsg_mem/bsg_nonsynth_mem_1r1w_sync_dma.v
 VSOURCES += $(BASEJUMP_STL_DIR)/bsg_mem/bsg_nonsynth_mem_1r1w_sync_mask_write_byte_dma.v
 VSOURCES += $(BASEJUMP_STL_DIR)/bsg_mem/bsg_nonsynth_mem_1rw_sync_mask_write_byte_dma.v
 
+# ???
```
Comment (mrutt92) [inline libraries/platforms/aws-vcs/hardware.mk:72]: Cache <-> Test DRAM Interface

Comment (drichmond) [reply libraries/platforms/aws-vcs/hardware.mk]: Done, thanks!
